most-adapter/src/index.ts

```diff
@@ -3,39 +3,78 @@ import {
   Observer,
   StreamSubscribe,
   DisposeFunction,
-  Subject,
+  Subject as CycleSubject,
 } from '@cycle/base';
 
-import {Stream} from 'most';
-import {subject} from 'most-subject';
+import {Stream, Sink, Disposable, Scheduler} from 'most';
+import {
+  BasicSubjectSource,
+  Subject as MostSubject,
+  subject as getMostSubject
+} from 'most-subject';
 import hold from '@most/hold';
 
+class AdaptedSubjectDisposable<T> implements Disposable<T> {
+  private disposed: boolean = false;
+  constructor (private subjectDisposable: Disposable<T>, private adaptedDisposer: DisposeFunction) {}
+
+  dispose () {
+    if (!this.disposed) {
+      this.disposed = true;
+      this.adaptedDisposer();
+      return this.subjectDisposable.dispose();
+    }
+  }
+}
+
+class AdaptedSubjectSource<T> extends BasicSubjectSource <T> {
+  constructor(protected originStream: any, protected originStreamSubscribe: StreamSubscribe) {
+    super();
+  }
+
+  run(sink: Sink<T>, scheduler: Scheduler): Disposable<T> {
+    const that = this;
+    let _dispose: DisposeFunction | void;
+    const _subjectDisposable = super.run(sink, scheduler);
+
+    setTimeout(() => {
+      _dispose = that.originStreamSubscribe(that.originStream, {
+        next: (x: T) => that.next(x),
+        error: (err: Error) => that.error(err),
+        complete: (x?: T) => {
+          that.complete(x);
+          if (typeof _dispose === 'function') {
+            _dispose();
+          }
+
+          _subjectDisposable.dispose();
+        }
+      });
+    }, 4);
+
+    return new AdaptedSubjectDisposable(_subjectDisposable, () => {
+      setTimeout(() => {
+        if (typeof _dispose === 'function') {
+          _dispose();
+        }
+      }, 4);
+    });
+  }
+}
+
 const MostAdapter: StreamAdapter = {
   adapt<T>(originStream: any, originStreamSubscribe: StreamSubscribe): Stream<T> {
     if (MostAdapter.isValidStream(originStream)) { return originStream; };
-    let dispose: any;
-    const stream = subject<any>();
-
-    dispose = originStreamSubscribe(originStream, {
-      next: (x: T) => stream.next(x),
-      error: (err: Error) => stream.error(err),
-      complete: (x?: T) => {
-        stream.complete(x);
-        if (typeof dispose === 'function') {
-          <DisposeFunction> dispose();
-        }
-      }
-    });
 
-    return stream;
+    return new MostSubject<T>(new AdaptedSubjectSource<T>(originStream, originStreamSubscribe));
```

Comment:
Why do we need `MostSubject` (and the custom source) here? Could we just use `[most.create](https://github.com/mostjs/create)` instead?